sklearn/model_selection/_search.py

```diff
@@ -784,7 +915,7 @@ def __init__(self, estimator, param_grid, scoring=None, fit_params=None,
             n_jobs=n_jobs, iid=iid, refit=refit, cv=cv, verbose=verbose,
             pre_dispatch=pre_dispatch, error_score=error_score)
         self.param_grid = param_grid
-        _check_param_grid(param_grid)
+        self.parameter_names_ = _check_param_grid_or_dist(param_grid)
```

Comment:
Ah, it seems that you are using it, you can extract it from `parameter_iterable[0]` in fit and set it to be a "private" attribute instead of doing the set update every time.